src/core/tensor/tensor_math_opencl.h

```diff
@@ -19,10 +19,1048 @@
 #ifndef  SINGA_CORE_TENSOR_TENSOR_MATH_OPENCL_H_
 #include "./tensor_math.h"
 
+// Kernel names
+constexpr const char* clkernel_abs = "clkernel_abs";
+
 namespace singa {
 
+// **************************************
+// Element-wise functions
+// **************************************
+
+template<>
+void Abs<float, lib::Opencl>(const size_t num, const Block* in, Block* out, Context* ctx) {
+  cl_int status = CL_SUCCESS;
+
+  std::string kname = "clkernel_abs";
+  auto kernel = oclDevice.GetKernel(kname);
```

Comment:
where is oclDevice defined?
better avoid global variables.
may get it from ctx?